ruma-identifiers/src/server_name.rs

```diff
@@ -1,7 +1,26 @@
 /// Check whether a given string is a valid server name according to [the specification][].
 ///
 /// [the specification]: https://matrix.org/docs/spec/appendices#server-name
-pub fn is_valid_server_name(name: &str) -> bool {
+use crate::error::Error;
+
+#[derive(Clone, Copy, Debug)]
+pub struct ServerName<T> {
+    full_id: T,
+}
+
+fn try_from<S, T>(server_name: S) -> Result<ServerName<T>, Error>
+where
+    S: AsRef<str> + Into<T>,
+{
+    if !is_valid_server_name(server_name.as_ref()) {
+        return Err(Error::InvalidServerName);
+    }
+    Ok(ServerName { full_id: server_name.into() })
+}
+
+common_impls!(ServerName, try_from, "An IP address or hostname");
+
+fn is_valid_server_name(name: &str) -> bool {
```

Comment:
Why is this function no longer public?

Reply:
It seemed to follow the pattern of the other identifiers to use `try_from` to verify the validity of the identifier.

Reply:
Let's keep it public since it was public before. Somebody might be relying on it. We could deprecate it though.